lib/utils.js

```diff
@@ -16,48 +19,107 @@ export const decodeBase64 = ifDefined((input) => localAtob(input.trim()));
+ * @param {String} str String to convert
+ * @returns {Uint8Array} An array of 8-bit integers
+ */
+export const binaryStringToArray = (str) => {
+    if (!isString(str)) {
+        throw new Error('binaryStringToArray: Data must be in the form of a string');
+    }
+
+    const result = new Uint8Array(str.length);
+    for (let i = 0; i < str.length; i++) {
+        result[i] = str.charCodeAt(i);
+    }
+    return result;
+};
+
+export const arrayToBinaryString = (bytes) => {
+    bytes = new Uint8Array(bytes);
+    const result = [];
+    const bs = 1 << 14;
+    const j = bytes.length;
+
+    for (let i = 0; i < j; i += bs) {
+        result.push(String.fromCharCode.apply(String, bytes.subarray(i, i + bs < j ? i + bs : j)));
+    }
+    return result.join('');
+};
+
 /**
- * SHA1 is an unsafe hash function. It should not be used for cryptographic purposes.
- * DO NOT USE in contexts where collision resistance is important
- * @see openpgp.crypto.hash.sha1
+ * Convert an array of 8-bit integers to a hex string
+ * @param {Uint8Array} bytes Array of 8-bit integers to convert
+ * @returns {String} Hexadecimal representation of the array
  */
-export const unsafeSHA1 = (args) => openpgp.crypto.hash.sha1(args);
+export const arrayToHexString = (bytes) => {
+    const r = [];
+    const e = bytes.length;
+    let c = 0;
+    let h;
+    while (c < e) {
+        h = bytes[c++].toString(16);
+        while (h.length < 2) {
+            h = '0' + h;
+        }
+        r.push('' + h);
+    }
+    return r.join('');
+};
 
 /**
  * Dearmor a pgp encoded message.
  * @param {String} input
  * @return {Promise<Uint8Array>}
  */
 export const stripArmor = async (input) => {
-    const { data } = await openpgp.armor.decode(input);
-    const bytes = await openpgp.stream.readToEnd(data);
+    const { data } = await unarmor(input);
+    const bytes = await readToEnd(data);
     return bytes;
 };
 
-export const getMaxConcurrency = () => {
-    const { workers = [null] } = openpgp.getWorker() || {};
-    return workers.length;
-};
+/**
+ * Convert a native javascript string to a Uint8Array of utf8 bytes
+ * @param {String|ReadableStream} str - The string to convert
+ * @returns {Uint8Array|ReadableStream} A valid squence of utf8 bytes.
+ */
+export function stringToUtf8Array(str) {
+    const encoder = new TextEncoder('utf-8');
+    function process(value, lastChunk = false) {
+        return encoder.encode(value, { stream: !lastChunk });
+    }
+    return transform(str, process, () => process('', true));
+}
 
-export const createWorker = ({ path = '', n }) => {
-    if (!path) {
-        throw new Error('Path to worker required');
+/**
+ * Convert a Uint8Array of utf8 bytes to a native javascript string
+ * @param {Uint8Array|ReadableStream} utf8 - A valid squence of utf8 bytes
+ * @returns {String|ReadableStream} A native javascript string.
+ */
+export function utf8ArrayToString(utf8) {
+    const decoder = new TextDecoder('utf-8');
```

Comment:
UTF-8 is the default now:
```suggestion
    const decoder = new TextDecoder();
```